snoop/data/analyzers/email.py

```diff
@@ -26,14 +27,56 @@ def get_headers(message):
     return dict(rv)
 
 
+def dump_part(message):
+    rv = {'headers': get_headers(message)}
+
+    content_type = message.get_content_type()
+    if content_type == 'text/plain':
+        payload_bytes = message.get_payload(decode=True)
+        charset = message.get_content_charset() or 'latin1'
+        rv['text'] = payload_bytes.decode(charset, errors='replace')
+
+    if content_type == 'text/html':
+        payload_bytes = message.get_payload(decode=True)
+        with models.Blob.create() as writer:
+            writer.write(payload_bytes)
+        rmeta_blob = tika.rmeta(writer.blob)
+        with rmeta_blob.open(encoding='utf8') as f:
+            rmeta_data = json.load(f)
+        rv['text'] = rmeta_data[0]['X-TIKA:content']
+
+    if message.is_multipart():
+        rv['parts'] = [dump_part(part) for part in message.get_payload()]
+
+    if message.get_content_disposition():
+        raw_filename = message.get_filename()
+        if raw_filename:
+            filename = str(email.header.make_header(
+                email.header.decode_header(raw_filename)
+            ))
+            attachment_content = message.get_payload(decode=True)
+
+            with models.Blob.create() as writer:
+                writer.write(attachment_content)
+
+            rv['attachment'] = {
+                'name': filename,
+                'blob_pk': writer.blob.pk,
+            }
+
+    return rv
+
+
 @shaorma('email.parse')
 def parse(blob):
     with blob.open() as f:
-        message = email.message_from_bytes(f.read())
+        message_bytes = f.read()
+
+    if message_bytes[:3] == b'\xef\xbb\xbf':
```

Comment:
What is this? Please include the words `byte order mark, thanks Microsoft` in a comment above this.